app/src/androidTest/java/com/android/voyageur/ui/overview/AddTripTest.kt

```diff
@@ -0,0 +1,93 @@
+import org.mockito.Mockito.*
+import org.mockito.kotlin.whenever
+
+class AddTripScreenTest {
+  private lateinit var tripRepository: TripRepository
+  private lateinit var navigationActions: NavigationActions
+  private lateinit var tripsViewModel: TripsViewModel
+
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @Before
+  fun setUp() {
+    tripRepository = mock(TripRepository::class.java)
+    navigationActions = mock(NavigationActions::class.java)
+    tripsViewModel = TripsViewModel(tripRepository)
+
+    `when`(navigationActions.currentRoute()).thenReturn(Screen.ADD_TRIP)
+    whenever(tripsViewModel.getNewTripId()).thenReturn("mockTripId")
+  }
+
+  private fun <T> anyNonNull(): T = Mockito.any<T>()
+
+  @Test
+  fun addTripScreen_initialState() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("inputTripTitle").assertExists()
+    composeTestRule.onNodeWithTag("inputTripDescription").assertExists()
+    composeTestRule.onNodeWithTag("inputTripCreator").assertExists()
+    composeTestRule.onNodeWithTag("inputTripParticipants").assertExists()
+    composeTestRule.onNodeWithTag("inputTripLocations").assertExists()
+    composeTestRule.onNodeWithTag("inputStartDate").assertExists()
+    composeTestRule.onNodeWithTag("inputEndDate").assertExists()
+    composeTestRule.onNodeWithTag("tripSave").assertExists()
+
+    composeTestRule.onNodeWithTag("addTripTitle").assertTextEquals("Create a New Trip")
+    composeTestRule.onNodeWithTag("tripSave").assertTextEquals("Save Trip")
+  }
+
+  @Test
+  fun addTripScreen_inputsUpdateState() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("inputTripTitle").performTextInput("London")
+    composeTestRule.onNodeWithTag("inputTripDescription").performTextInput("4 days in London")
+    composeTestRule.onNodeWithTag("inputTripCreator").performTextInput("John Doe")
+    composeTestRule.onNodeWithTag("inputTripParticipants").performTextInput("Alice, Bob")
+
+    composeTestRule.onNodeWithTag("inputTripTitle").assertExists()
+    composeTestRule.onNodeWithTag("inputTripDescription").assertExists()
+    composeTestRule.onNodeWithTag("inputTripCreator").assertExists()
+    composeTestRule.onNodeWithTag("inputTripParticipants").assertExists()
+  }
+
+  @Test
+  fun doesNotSubmitWithInvalidDate() {
+    composeTestRule.setContent { AddTripScreen(tripsViewModel, navigationActions) }
+
+    composeTestRule.onNodeWithTag("inputStartDate").performTextClearance()
+    composeTestRule.onNodeWithTag("inputStartDate").performTextInput("notadate")
+
+    composeTestRule.onNodeWithTag("inputEndDate").performTextClearance()
+    composeTestRule.onNodeWithTag("inputEndDate").performTextInput("05/01/2024")
+
+    composeTestRule.onNodeWithTag("tripSave").performClick()
+
+    verify(tripRepository, never()).updateTrip(anyNonNull(), anyNonNull(), anyNonNull())
+  }
+
```

Comment:
Here you could also add a test to check that the user can't submit with an empty name for the trip.